### PR TITLE
Update documentation for automatic model undeployment

### PR DESCRIPTION
**Pull request by Zhangxunmt**
### Description
Update the Model Undeploy API to include the automatic undeploy feature based on TTL for all ML models in ML-Commons.

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/7061

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thank you, @Zhangxunmt ! Can you please address the Vale errors? I'll send forward for review.

File: _ml-commons-plugin/api/model-apis/undeploy-model.md [hunk 1]
```diff
@@ -56,3 +56,13 @@ POST /_plugins/_ml/models/_undeploy
   }
 }
 ```
+### Automatically undeploy a model based on TTL
+
+Starting with OpenSearch version 2.14, models can be automatically undeployed from memory based on the pre-defined TTL (time-to-live) since they were last accessed or used. To define a TTL that auto undeploys a model, include the below ModelDeploySetting in your ML mode.
```
Comment (Naarcha-AWS) [inline _ml-commons-plugin/api/model-apis/undeploy-model.md:61]:
```suggestion
Starting with OpenSearch  2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL), when the model was last accessed or used.. To define a TTL that auto undeploys a model, include the following `ModelDeploySetting` in your ML mode.
```

File: _ml-commons-plugin/remote-models/index.md [hunk 2]
```diff
@@ -322,6 +322,10 @@ The response contains the inference results provided by the OpenAI model:
 
 To learn how to use the model for vector search, see [Using an ML model for neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/#using-an-ml-model-for-neural-search).
 
+## Step 7 (Optional): Undeploy the model 
+
+Undeploy the model automatically by defining a TTL in the Model or use the Undeploy API to undeploy the model manually. To learn more about Model undeploy, see the [Undeploy API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/undeploy-model/).
```
Comment (Naarcha-AWS) [inline _ml-commons-plugin/remote-models/index.md:327]:
```suggestion
Undeploy the model automatically by defining a TTL in the model settings or use the Undeploy API to undeploy the model manually. To learn more about model undeploy, see [Undeploy API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/undeploy-model/).
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by Zhangxunmt** [comment]
@Naarcha-AWS , I have addressed the comments in the refresh. Please review and approve/merge accordingly!

File: _ml-commons-plugin/api/model-apis/undeploy-model.md [hunk 3]
```diff
@@ -56,3 +56,13 @@ POST /_plugins/_ml/models/_undeploy
...
 }
 ```
+### Automatically undeploy a model based on TTL
+
+Starting with OpenSearch version 2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following ModelDeploySetting in your ML mode.
+
+```json
+PUT /_plugins/_ml/models/COj7K48BZzNMh1sWedLK
+{
+    "deploy_setting": {"model_ttl_minutes" : 100}
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/undeploy-model.md:66]: User should update model TTL when model undeployed.

File: _ml-commons-plugin/api/model-apis/undeploy-model.md [hunk 4]
```diff
@@ -56,3 +56,35 @@ POST /_plugins/_ml/models/_undeploy
...
+
+#### Example request: Creating a model with TTL
+```json
+POST /_plugins/_ml/models/_register
+ {
+   "name": "Sample Model Name",
+   "function_name": "remote",
+   "description": "test model",
+   "connector_id": "-g1nOo8BOaAC5MIJ3_4R",
+   "deploy_setting": {"model_ttl_minutes": 100}
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/undeploy-model.md:80]: Can user use this for local model?

File: _ml-commons-plugin/api/model-apis/undeploy-model.md [hunk 5]
```diff
@@ -56,3 +56,35 @@ POST /_plugins/_ml/models/_undeploy
   }
 }
 ```
+### Automatically undeploy a model based on TTL
+
+Starting with OpenSearch version 2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following ModelDeploySetting in your ML mode. Please note that model TTLs are checked periodically by a syn_up cron job, so the maximum time a model lives in the memory could be TTL + sync_up_job_interval. The default cron job interval is 10 seconds. To update the cron job internal, use the following cluster setting.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/undeploy-model.md:61]: Check the `GitHub Actions / style-job` 
> Using 'Please' is unnecessary. Remove

Comment (ylwu-amzn) [reply _ml-commons-plugin/api/model-apis/undeploy-model.md:61]: If I'm understanding correctly, the user can indeed update the model's Time To Live (TTL) after it has been deployed without encountering an exception. However, this TTL update will not take effect immediately. A deployment or redeployment of the model is required for the updated TTL to come into effect.

File: _ml-commons-plugin/api/model-apis/undeploy-model.md (hunk 4 above)
Comment (Zhangxunmt) [reply _ml-commons-plugin/api/model-apis/undeploy-model.md:80]: Yes it's for all models. Only local model require manual deploy. Everything else is automated in terms of model deploy/un-deploy.

**Review by ylwu-amzn — APPROVED** [review]

File: _ml-commons-plugin/api/model-apis/undeploy-model.md [hunk 6]
```diff
@@ -56,3 +56,35 @@ POST /_plugins/_ml/models/_undeploy
   }
 }
 ```
+### Automatically undeploy a model based on TTL
+
+Starting with OpenSearch version 2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following ModelDeploySetting in your ML mode. Note that model TTLs are checked periodically by a syn_up cron job, so the maximum time a model lives in the memory could be TTL + sync_up_job_interval. The default cron job interval is 10 seconds. To update the cron job internal, use the following cluster setting.
```
Comment (Naarcha-AWS) [inline _ml-commons-plugin/api/model-apis/undeploy-model.md:61]:
```suggestion
Starting with OpenSearch  2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following `ModelDeploySetting` in your ML node. Note that model TTLs are checked periodically by a `syn_up` cron job, so the maximum time a model lives in the memory could be TTL + the `sync_up_job_` interval. The default cron job interval is 10 seconds. To update the cron job internally, use the following cluster setting:
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

File: _ml-commons-plugin/api/model-apis/undeploy-model.md [hunk 7]
```diff
@@ -56,3 +56,35 @@ POST /_plugins/_ml/models/_undeploy
   }
 }
 ```
+### Automatically undeploy a model based on TTL
+
+Starting with OpenSearch  2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following `ModelDeploySetting` in your ML node. Note that model TTLs are checked periodically by a `syn_up` cron job, so the maximum time a model lives in the memory could be TTL + the `sync_up_job_` interval. The default cron job interval is 10 seconds. To update the cron job internally, use the following cluster setting:
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/undeploy-model.md:61]:
```suggestion
Starting with OpenSearch 2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that automatically undeploys a model, include the following `ModelDeploySetting` in your machine learning (ML) node. Note that model TTLs are checked periodically by a `syn_up` cron job, so the maximum time that a model lives in memory could be TTL + the `sync_up_job_` interval. The default cron job interval is 10 seconds. To update the cron job internally, use the following cluster setting:
```

File: _ml-commons-plugin/remote-models/index.md [hunk 8]
```diff
@@ -322,6 +322,10 @@ The response contains the inference results provided by the OpenAI model:
 
 To learn how to use the model for vector search, see [Using an ML model for neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/#using-an-ml-model-for-neural-search).
 
+## Step 7 (Optional): Undeploy the model 
+
+Undeploy the model automatically by defining a TTL in the model settings or use the Undeploy API to undeploy the model manually. To learn more about model undeploy, see [Undeploy API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/undeploy-model/).
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/index.md:327]:
```suggestion
You can undeploy the model automatically by defining a TTL in the model settings or by using the Undeploy API to undeploy the model manually. For more information, see [Undeploy API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/undeploy-model/).
```

File: _ml-commons-plugin/api/model-apis/undeploy-model.md [hunk 9]
```diff
@@ -56,3 +56,35 @@ POST /_plugins/_ml/models/_undeploy
...
+ {
+   "name": "Sample Model Name",
+   "function_name": "remote",
+   "description": "test model",
+   "connector_id": "-g1nOo8BOaAC5MIJ3_4R",
+   "deploy_setting": {"model_ttl_minutes": 100}
+ }
+```
+
+#### Example request: Updating a model with TTL when the model is undeployed
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/undeploy-model.md:84]:
```suggestion
#### Example request: Updating a model with a TTL when the model is undeployed
```

File: _ml-commons-plugin/api/model-apis/undeploy-model.md [hunk 10]
```diff
@@ -56,3 +56,35 @@ POST /_plugins/_ml/models/_undeploy
...
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.sync_up_job_interval_in_seconds": 10
+    }
+}
+```
+
+#### Example request: Creating a model with TTL
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/undeploy-model.md:72]:
```suggestion
#### Example request: Creating a model with a TTL
```

**Review by natebower — COMMENTED** [review]
@Zhangxunmt @Naarcha-AWS Please see my changes and let me know if you have any questions. Thanks!

**Comment by Zhangxunmt** [comment]
@natebower @Naarcha-AWS Please check my latest commit. It should have addressed all your comments. Please approve and merge when ready.

**Pull request merged**